### DISCUSSION TITLE
`dependentRequired` in schema validation

### DISCUSSION THREAD

**Discussion by onosendi** [fastify/fastify#3613] [Q&A] [answered]
This may not be the correct place for this. If not, I apologize in advance.

[dependentRequired](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.5.4) seems to have no affect:

```js
    properties: {
      currentPassword: { type: 'string' },
      password: { $ref: 'users#/properties/password' },
      passwordConfirm: { $ref: 'users#/properties/passwordConfirm' },
    },
    dependentRequired: {
      currentPassword: ['password', 'passwordConfirm'],
      password: ['currentPassword', 'passwordConfirm'],
      passwordConfirm: ['currentPassword', 'password'],
    },
```

**Comment by mcollina** [comment, marked as answer]
Have you checked if this works in Ajv?

**Reply by onosendi** [reply]
I guess it doesn't.

**Reply by Eomm** [reply]
It should

> Out of the box, Ajv supports JSON Schema (drafts 04, 06, 07, 2019-09 and 2020-12)

Assuming you are using fastify v3, you need to upgrade ajv by yourself:
https://www.nearform.com/blog/upgrading-fastifys-input-validation-to-ajv-version-8/

**Comment by MoSattler** [comment]
Any news on this? I get 
```
FastifyError [Error]: Failed building the validation schema for GET: /rest/:siteId/trackers, due to error strict mode: unknown keyword: "dependentRequired"
```

**Reply by onosendi** [reply]
https://www.nearform.com/insights/upgrading-fastifys-input-validation-to-ajv-version-8/
